### PR TITLE
60584 - Set partner as required when journal is sale or expense

### PR DESCRIPTION
**Pull request by paa-axelor**

### CODE REVIEW

File: axelor-account/src/main/resources/views/MoveTemplateLine.xml [hunk 1]
```diff
@@ -4,7 +4,8 @@
   xsi:schemaLocation="http://axelor.com/xml/ns/object-views http://axelor.com/xml/ns/object-views/object-views_5.4.xsd">
 
   <grid name="move-template-line-in-amount-grid" title="Move template line"
-    model="com.axelor.apps.account.db.MoveTemplateLine">
+    model="com.axelor.apps.account.db.MoveTemplateLine"
+    onNew="action-attrs-move-template-line-set-required-partner">
```
Comment (jmi-axelor) [inline axelor-account/src/main/resources/views/MoveTemplateLine.xml:8]: we need to talk about this feature with Geoffrey, I'm not sure we really want to do this in AOS. 
Also I don't understand why in "move-template-line-in-amount-form" there is a partner but not in "move-template-line-in-amount-grid".
If you call "action-attrs-move-template-line-set-required-partner" on the "move-template-line-in-amount-grid" onNew, I'm not sure how the view will behave

**Review by jmi-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/resources/views/MoveLine.xml [hunk 2]
```diff
@@ -823,6 +823,7 @@
 
   <action-group name="action-account-move-line-onnew-group">
     <action name="action-move-line-record-get-move-informations"/>
+    <action name="action-attrs-move-line-set-required-partner"/>
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/resources/views/MoveLine.xml:826]: Added

File: axelor-account/src/main/resources/views/Move.xml [hunk 3]
```diff
@@ -98,7 +98,8 @@
           <field name="partner" colSpan="6" canNew="false" canEdit="false"
             readonlyIf="$validatePeriod || moveLineList.length &gt; 0" form-view="partner-form"
             grid-view="partner-grid" onSelect="action-account-move-method-filter-partner"
-            onChange="action-move-group-partner-onchange"/>
+            onChange="action-move-group-partner-onchange"
+            requiredIf="$contains([1,2], journal.journalType.technicalTypeSelect)"/>
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/resources/views/Move.xml:102]: Action-attrs removed

File: axelor-account/src/main/resources/views/MoveTemplateLine.xml [hunk 4]
```diff
@@ -32,10 +33,13 @@
...
   <form name="move-template-line-in-amount-form" title="Move template line"
-    model="com.axelor.apps.account.db.MoveTemplateLine">
+    model="com.axelor.apps.account.db.MoveTemplateLine"
+    onNew="action-attrs-move-template-line-set-required-partner"
+    onLoad="action-attrs-move-template-line-set-required-partner">
     <panel name="mainPanel">
       <field name="name"/>
-      <field name="partner" canEdit="false" form-view="partner-form" grid-view="partner-grid"/>
+      <field name="partner" canEdit="false" form-view="partner-form" grid-view="partner-grid"
+        onSelect="action-attrs-move-template-line-set-required-partner"/>
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/resources/views/MoveTemplateLine.xml]: Removed

File: axelor-account/src/main/resources/views/MoveTemplateLine.xml (hunk 1 above)
Comment (paa-axelor) [reply axelor-account/src/main/resources/views/MoveTemplateLine.xml:8]: Added a verification on action-attrs to verify parent.Model

**Review by jmi-axelor — APPROVED** [review]

**Comment by jmi-axelor** [comment]
Merged in aos-sigma

**Pull request closed without merging**